### ISSUE TITLE
Unable to render when using with NextJS v13 & React Server Components

### ISSUE THREAD

**Issue by nick-kang** [vanilla-extract-css/vanilla-extract#929]
### Describe the bug

App does not render when I import  a `.css.ts` file in a React Server Component. See [component](https://github.com/nick-kang/vanilla-extract-repro-001/blob/main/app/page.tsx) in repro repo.

Workaround is to convert the `.css.ts` to a default export and use [swc-plugin-vanilla-extract](https://github.com/kwonoj/swc-plugin-vanilla-extract) instead of `@vanilla-extract/next-plugin`

```
// page.css.ts before
export const container = style({
  background: 'blue'
})
```

```
// page.css.ts workaround
const container = style({
  background: 'blue'
})

export default {
  container
}
```

### Reproduction

https://github.com/nick-kang/vanilla-extract-repro-001

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (4) x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
    Memory: 29.65 GB / 46.75 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 18.7.0 - /run/user/1000/fnm_multishells/553958_1669586582142/bin/node
    Yarn: 3.2.2 - /run/user/1000/fnm_multishells/553958_1669586582142/bin/yarn
    npm: 8.15.0 - /run/user/1000/fnm_multishells/553958_1669586582142/bin/npm
    Watchman: 2022.07.04.00 - /home/linuxbrew/.linuxbrew/bin/watchman
  Browsers:
    Chrome: 107.0.5304.121
    Firefox: 107.0
  npmPackages:
    @vanilla-extract/css: ^1.9.2 => 1.9.2 
    @vanilla-extract/next-plugin: ^2.1.1 => 2.1.1 
    @vanilla-extract/webpack-plugin: ^2.2.0 => 2.2.0
```

### Used Package Manager

pnpm

### Logs

```shell
TypeError: Cannot read properties of undefined (reading 'chunks')
    at getCssInlinedLinkTags (/node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:362:49)
    at createComponentTree (/node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:630:52)
    at /node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:753:62
    at Array.map (<anonymous>)
    at createComponentTree (/node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:717:84)
    at /node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:753:62
    at Array.map (<anonymous>)
    at createComponentTree (/node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:717:84)
    at /node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:753:62
    at Array.map (<anonymous>)
    at createComponentTree (/node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:717:84)
    at /node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:753:62
    at Array.map (<anonymous>)
    at createComponentTree (/node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:717:84)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:753:56
    at async Promise.all (index 0)
    at async createComponentTree (/node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:717:38)
    at async /node_modules/.pnpm/next@13.0.5_biqbaboplfbrettd7655fr4n2y/node_modules/next/dist/server/app-render.js:993:51
```
```


### Validations

- [X] Check that there isn't [already an issue](https://github.com/vanilla-extract-css/vanilla-extract/issues) that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by audifaxdev** [comment]
@nick-kang I've tried your code and the suggested workaround, but I could not get the blue background style to apply.

 I expect the background of the container to be blue, but the class style is missing and I get a 404 on http://localhost:3000/_next/app/page:app/page

**Comment by hamlim** [comment]
I think that the linked swc plugin only emulates the previous babel plugin / test transform and doesn't actually create the css resource.

**Comment by lilingxi01** [comment]
Is maintainers of this library planning to support Next.js 13 (appDir) and React Server Component?

**Comment by mdecrocemovson** [comment]
@nick-kang what does your next.config.js look like with this workaround? Trying to use the same one.

**Comment by pr-o** [comment]
I'm having the same problem.

gonna sit here and wait on Next.js 13 (+ appDir) supports 👀

**Comment by bizarre** [comment]
Two small patches for NextJS + VE NextJS plugin for anybody who really wants to use VE w/ the appDir beta before official support: https://gist.github.com/bizarre/825cab8224c28e93ff1a3933642f9271 (use with [patch-package](https://www.npmjs.com/package/patch-package)).

Only caveat I've found so far is that `globalStyle` doesn't seem to actually be global for server components, otherwise works pretty well.

**Comment by graup** [comment]
@bizarre I tested your patch and it works for me as well. I don't see any problem with globalStyle.

Do you want to submit a PR for VE? Not sure what we should do about the `mod.default` part in nextjs' code, maybe submit a PR there too?

**Comment by madjam002** [comment]
@bizarre This patch works great for me, although I've noticed some strange behaviour where if I have a .css.ts file that uses vanilla-extract/sprinkles and I have lots of properties passed to `defineProperties`, no CSS is output when doing a `next build` even though the class names appear in the markup. `next dev` works fine though.

If I reduce the number of properties passed to `defineProperties`, it works fine. I'm guessing there is some kind of limit I'm hitting with how many CSS classes are being generated. Planning to dig a little deeper to find out what this could be.

**Comment by chriskrogh** [comment]
Any updates here? Looking forward to using VE with RSC 😄

**Comment by NicoZweifel** [comment]
Using styled mdx pages as ServerComponents and [`@storybook/nextjs`](https://www.npmjs.com/package/@storybook/nextjs) and everything seems to be fine with  `dev` and `storybook` but am  experiencing some issues with `build` , `globalStyle` and `createGlobalTheme` like mentioned by others previously.

Currently on [`next@13.1.3`](https://www.npmjs.com/package/next/v/13.1.3) using an adjusted [version](https://gist.github.com/NicoZweifel/13991102a97d53524dfab5868c812444) of @bizarre 's [patch](https://gist.github.com/bizarre/825cab8224c28e93ff1a3933642f9271) with [patch-package](https://www.npmjs.com/package/patch-package) and appDir: true in next.config.js.
```js
  experimental: {
    appDir: true,
  },
```
 The issue of `globalStyle ` and `createGlobalTheme` not actually beeing global for server components still persists.

@madjam002 Observing the same behaviour still with large chunks from sprinkles. Splitting up the property definitions and sprinkles solved it, except for the color options which seem to be too many classes generated - no matter which way I split it.

@graup  There are some changes in app.render.js and checking for mod.default does not do anything for me anymore since it seems to be always defined. I was thinking about submitting a PR to VE, but appDir won't work without the regex fix in next and it will crash with `chunks undefined` or sth similar like in OP's post.


As of now able to work fine in 'dev' and 'storybook', but would be happy to work on a PR or come back to this issue later when upgrading dependencies.

**Comment by bradcerasani** [comment]
Setting up Next.js 13 w/appDir and Vanilla Extract today and ran into this as well. 

[Updated the above patches](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f) based on current versions (Next.js 13.2.4, Vanilla Extract Next Plugin 2.1.1).

If you haven't used [`patch-package`](https://www.npmjs.com/package/patch-package) before:

1. Download Zip of [gist](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f)
2. Rename unzipped folder to "patches", then move to project root
3.  Add `"postinstall": "patch-package"` to the `scripts` key in `package.json`
4. `npm i patch-package`

**Comment by thiagop90** [comment]
> Configurando o Next.js 13 w / appDir e Vanilla Extract hoje e se deparou com isso também.
> 
> [Atualizado os patches acima](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f) com base nas versões atuais (Next.js 13.2.4, Vanilla Extract Next Plugin 2.1.1).
> 
> Se você ainda não usou [`o patch-package`](https://www.npmjs.com/package/patch-package) antes:
> 
> 1. Baixar Zip of [gist](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f)
> 2. Renomeie a pasta descompactada para "patches" e, em seguida, mova para a raiz do projeto
> 3. Adicionar à chave em `"postinstall": "patch-package"``scripts``package.json`
> 4. `npm i patch-package`

I downloaded the zip and did as you said but I never used the patch-package, how to use it to install the patches?

**Comment by christiankaindl** [comment]
@thiagop90: the steps that @bradcerasani provided are all you need. You should be able to see the patches being applied in the terminal after running `npm install`

![obfuscated-gh](https://user-images.githubusercontent.com/15364860/229743299-4a816689-03c8-4d14-837b-10bc9892efaa.png)

You may need to update your package versions to match those from the patches

**Comment by christiankaindl** [comment]
@bradcerasani you patch works great when running locally, but when I build the app the styles don't get applied. Any idea why that could be? Did you run into the issue as well?

**Comment by trm217** [comment]
Does it still work with `next@13.3.0` for y'all?

**Comment by airtonix** [comment]
> Setting up Next.js 13 w/appDir and Vanilla Extract today and ran into this as well.
> 
> [Updated the above patches](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f) based on current versions (Next.js 13.2.4, Vanilla Extract Next Plugin 2.1.1).
> 
> If you haven't used [`patch-package`](https://www.npmjs.com/package/patch-package) before:
> 
> 1. Download Zip of [gist](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f)
> 2. Rename unzipped folder to "patches", then move to project root
> 3. Add `"postinstall": "patch-package"` to the `scripts` key in `package.json`
> 4. `npm i patch-package`

people still use that? 

i thought everyone moved onto yarn 4 when they handed in their rotary dial phones?

**Comment by trm217** [comment]
@airtonix what a way to enrich the GitHub community, way to go! 👏🏼🙄

**Comment by samithaf** [comment]
> Setting up Next.js 13 w/appDir and Vanilla Extract today and ran into this as well.
> 
> [Updated the above patches](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f) based on current versions (Next.js 13.2.4, Vanilla Extract Next Plugin 2.1.1).
> 
> If you haven't used [`patch-package`](https://www.npmjs.com/package/patch-package) before:
> 
[@vanilla-extract__next-plugin@2.1.1.patch](https://github.com/vanilla-extract-css/vanilla-extract/files/11185353/%40vanilla-extract__next-plugin%402.1.1.patch)
[next@13.3.0.patch](https://github.com/vanilla-extract-css/vanilla-extract/files/11185354/next%4013.3.0.patch)

> 1. Download Zip of [gist](https://gist.github.com/bradcerasani/633b55a2b89af5b2dd7f73cfecefad9f)
> 2. Rename unzipped folder to "patches", then move to project root
> 3. Add `"postinstall": "patch-package"` to the `scripts` key in `package.json`
> 4. `npm i patch-package`

above patch works and if anyone who is using pnpm here are the patch files.

**Comment by samkingco** [comment]
Also running into the same issues. Patches work fine and `next dev` works as expected, however when running `next build` and `next start`, none of my styles exist. The class names are there in the server component markup, just no link to the CSS file in the head, thus no style is applied. I can see the CSS files exist in `.next/static/css` and they reference class names in the server component, it's just not being linked to in the layout code.

**Comment by airtonix** [comment]
> @airtonix what a way to enrich the GitHub community, way to go! 👏🏼roll_eyes

glad to help.

**Comment by SuttonJack** [comment]
https://github.com/vercel/next.js/pull/48306

**Comment by SuttonJack** [comment]
This issue should be resolved on `next@13.3.1-canary.6` (thanks to @shuding on https://github.com/vercel/next.js/pull/48306)

<h1></h1>

Working example repo [here](https://github.com/SuttonJack/vanilla-extract-app-dir)

**Comment by graup** [comment]
Thanks @SuttonJack! Your example repo works great for me, both dev and build. [Export](https://nextjs.org/docs/advanced-features/static-html-export) also works if you remove `export const runtime = 'experimental-edge'` from page.tsx.

I'll go ahead and close this issue. If anyone encounters any new problems, please open a new issue!

**Comment by KhachGevorgyan94** [comment]
Hi everyone!
This   my computer options

   OS: windows 11
    CPU: 12th x64 Intel(R) Core(TM) i9-12900HCPU
    Memory: 16GB / DDR5
  Binaries:
    Node: v18.13.0
    Yarn: v1.22.18
    npm: v8.19.3
  Browsers:
    Chrome: Version 112.0.5615.121
  npmPackages:
    "@types/node": "18.15.11",
    "@types/react": "18.0.34",
    "@types/react-dom": "18.0.11",
    "@vanilla-extract/css": "^1.11.0",
    "@vanilla-extract/next-plugin": "^2.1.2",
    "typescript": "5.0.4",
    
    
![image](https://user-images.githubusercontent.com/53270475/233021465-63daf12e-52bd-41bc-8cc9-b15d5e9a19e9.png)
![image](https://user-images.githubusercontent.com/53270475/233022216-2bcbff0e-aca2-4245-b0d2-4ae7358512d6.png)

I run the project but no style is visible
this project <a href="https://github.com/SuttonJack/vanilla-extract-app-dir">here</a>

What is the reason that style-s do not work on windows 11,but on Mac it's completely fine

**Comment by SuttonJack** [comment]
@KhachGevorgyan94 - can you send a screenshot of the errors in your console?

**Comment by KhachGevorgyan94** [comment]
![image](https://user-images.githubusercontent.com/53270475/233036442-838fb640-e67e-4c0a-bc98-75f1435ff254.png)

This is a google chrome bugs

**Comment by zackshackleton** [comment]
While using the canary builds, I notice styles are coming through correctly, however our sprinkles are not. Classnames are being attached to the markup but the css is not there. 

This is a large application with a lot of sprinkles. While building the project or running in dev mode, I get quite a few serialization warnings due to the sprinkles as well.

**Comment by KhachGevorgyan94** [comment]
I realized that it still doesn't work for windows operating system and it has flaws, yesterday I tried it on linux operating system and it worked normally․
I will ask specialists to pay attention to this global problem.
Thanks

**Comment by e2goon** [comment]
It exposes the class name correctly on Windows 11, but it doesn't work properly because there is no CSS. I'm having the same issue.

**Comment by jungpaeng** [comment]
I'd like to know the status of this issue.

**Comment by juice49** [comment]
I ran into some problems getting Vanilla Extract to work with Next.js version `13.4.1`. It took me a little while to figure out, but Vanilla Extract will only work when `experimental.appDir` is set to `true` in `next.config.js`. Because I'm using a Next.js version in which the App Router is stable, I didn't have this option enabled. After I switched it on, Vanilla Extract started working!

```ts
const { createVanillaExtractPlugin } = require('@vanilla-extract/next-plugin')

const withVanillaExtract = createVanillaExtractPlugin()

/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    // The App Router is stable, but I ran into issues getting Vanilla Extract
    // to work without including this option.
    appDir: true,
  },
}

module.exports = withVanillaExtract(nextConfig)

```

**Comment by adri1wald** [comment]
@juice49 it works in dev now but when I do a production build it fails spectacularly.

**Comment by juice49** [comment]
@adri1wald I just tested a production build and it's working for me, so there must be some other issues at play for you 😔. What errors do you see?

**Comment by adri1wald** [comment]
@juice49 Should've clarified it's effectively what people above are saying. classes and vars are all correct in the markup, but the actual css is missing.

**Comment by juice49** [comment]
@adri1wald Ah, I'm sorry. I should also clarify that I'm using macOS. Based on reports from other folks, it seems like there could be an issue when running in Windows. If your example project code is available, I can see what happens when I try to build it.

**Comment by adri1wald** [comment]
I run macOS too and this also failed in the vercel cloud

**Comment by juice49** [comment]
I just tested a Vercel build, and everything seems to be working correctly for me.

**Comment by graup** [comment]
It seems there are a few different issues here (1. builds not outputting css on Windows, 2. not working without the experimental.appDir setting). Could someone open new issues for these? I have a hunch that both of those will require help from the Next.js side.

**Comment by adri1wald** [comment]
Here's a [link to my repo](https://github.com/adri1wald/mrlowlevel/tree/next-13). Maybe you can see what is wrong?

**Comment by daformat** [comment]
There also seem to be a 3rd issue with the **css being outputted in the build, but not included in the page**, this happens with sprinkles when there are too many output classes, yet I am unsure if this is sprinkles specific, or just that sprinkles tend to increase faster than the rest.

- Here's an example that fails: https://vanilla-extract-next-app-dir-mono-4xgghql5a-daformat.vercel.app/

- Here's one that doesn't: https://vanilla-extract-next-app-dir-mono-ote39xyrd-daformat.vercel.app/

The simultaneous usage of two different sets of sprinkles (defined separately in different .css.ts files) is the only difference, with the faulty outputted css going from 8kb to 16kb and not being included in the page anymore

- Related commit (from working to not working): https://github.com/daformat/vanilla-extract-next-app-dir-mono/commit/263819e1dfa3d381e20e0940fe67cce22493ff32
- Reproduction repo: https://github.com/daformat/vanilla-extract-next-app-dir-mono

This feels similar to what [madjam002 described](https://github.com/vanilla-extract-css/vanilla-extract/issues/929#issuecomment-1352927778)

**Comment by exneval** [comment]
> Could someone open new issues for these? I have a hunch that both of those will require help from the Next.js side.

Done

**Comment by i-bsd** [comment]
> I ran into some problems getting Vanilla Extract to work with Next.js version `13.4.1`. It took me a little while to figure out, but Vanilla Extract will only work when `experimental.appDir` is set to `true` in `next.config.js`. Because I'm using a Next.js version in which the App Router is stable, I didn't have this option enabled. After I switched it on, Vanilla Extract started working!
> 
> This is because [the `hasAppDir` check here](https://github.com/vanilla-extract-css/vanilla-extract/blob/0d0ea3909e7f952f24aafa4c9653853ac5841b8c/packages/next-plugin/src/index.ts#L51) relies on the `experimental.appDir` option being set.
> 
> ```ts
> const { createVanillaExtractPlugin } = require('@vanilla-extract/next-plugin')
> 
> const withVanillaExtract = createVanillaExtractPlugin()
> 
> /** @type {import('next').NextConfig} */
> const nextConfig = {
>   experimental: {
>     // The App Router is stable, but the Vanilla Extract Next Plugin
>     // currently requires that `experimental.appDir` is `true`.
>     //
>     // https://github.com/vanilla-extract-css/vanilla-extract/issues/929#issuecomment-1538555608
>     appDir: true,
>   },
> }
> 
> module.exports = withVanillaExtract(nextConfig)
> ```
> 
> @e2goon You're using the same Next.js version as me. Maybe this will fix the problem for you.

Thanks for this. Works with experimental appDir flag. Curious though why after many weeks there's still no vanilla-extract update to remove this requirement.

**Comment by graup** [comment]
We're waiting for feedback from the Next.js team. It's not clear how to fix this on the Vanill Extract end. https://github.com/vercel/next.js/discussions/49892

**Comment by i-bsd** [comment]
Also, the CSS not appearing in builds isn't just a Windows issue. Happening for me on Ubuntu also.

**Comment by jere-co** [comment]
@i-bsd  Try this for now https://github.com/vanilla-extract-css/vanilla-extract/issues/1085#issuecomment-1555986222

**Comment by thevisioner** [comment]
> @i-bsd & @thevisioner Try this for now [#1085 (comment)](https://github.com/vanilla-extract-css/vanilla-extract/issues/1085#issuecomment-1555986222)

Unfortunately, this solution doesn't work for me either.

**Comment by bgfernandes** [comment]
Having a similar issue, running ubuntu, on dev, everything works fine, when doing a production build, CSS file is not linked in the HTML.

I tried the solution @jere-co pointed it worked for me: 


needs to disable splitChunks in the next.config.js:
```
const nextConfig = {
  experimental: {
    appDir: true,
  },
  webpack: config => {
    config.optimization.splitChunks = false;

    return config;
  },
};
```

**Comment by joshua7v** [comment]
I'm on the same boat with @thevisioner 
with exact the same config @jere-co provides

```js
const nextConfig = {
  experimental: {
    appDir: true,
  },
  webpack: config => {
    config.optimization.splitChunks = false;

    return config;
  },
};
```

It seems the issue is Windows only. 
I can see classes are applied correctly, but no styles at all in both dev/production mode

I have to use page routes instead of app dir for now. old page routes does not have the issue.

**Comment by csi-lk** [comment]
> issue is Windows only

**Comment by mubarakone** [comment]
Any updates on this issue? @thevisioner and @joshua7v  have you found any solutions? This is a very complicated issue that is affecting tons of Next.js 13 devs it seems like.

**Comment by thevisioner** [comment]
Hi!Unfortunately, due to the great lack of specificity, I've completely moved away from trying CSS-in-JS at the moment. For me personally, full compatibility with the React Server Component model is important and that's why I've gone back to the basics - using CSS Modules. It is also noticeable that throughout the ecosystem many of the solutions are going back to the basics/standards. I don't know if it will be the same with CSS, but in a sense, I myself am willing to accept this path. From: Mubarak UsmaneSent: 05 September 2023 01:38To: vanilla-extract-css/vanilla-extractCc: Raimonds Sarkanbardis; MentionSubject: Re: [vanilla-extract-css/vanilla-extract] Unable to render when using with NextJS v13 & React Server Components (Issue #929) Any updates on this issue? @thevisioner and @joshua7v have you found any solutions? This is a very complicated issue that is affecting tons of Next.js 13 devs it seems like.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

**Comment by christiankaindl** [comment]
@thevisioner I am using it on mac and linux without problems--without using a custom next.config.js

**Comment by Dremora** [comment]
@thevisioner I don't see how CSS Modules solve the specificity issue. The loading order of CSS Modules is still handled by a bundler (e.g. Webpack).
To solve specificity issues with v-e, I recommend using [layers API](https://vanilla-extract.style/documentation/api/layer/) (which is a [web standard](https://developer.mozilla.org/en-US/docs/Web/CSS/@layer)). The docs don't seem to mention it, but `defineProperties` from Sprinkles allows passing `@layer` where sprinkles are generated. This way, you can mix component-specific and global (Sprinkles) styles without worrying about the order: Sprinkles will always have a lower specificity.

**Comment by thevisioner** [comment]
@Dremora I seem to have been vague. I don't mean the specificity of CSS, but rather the specifics (lack of clarity) in the context of the RSC compatibility issues of various CSS-in-JS solutions. And for now, RSC compatibility is one of my requirements.

**Comment by syfxlin** [comment]
I've recently been in the process of migrating my blog from Gatsby to Next.js, and I'm also experiencing problems with the Next.js RSC not outputting CSS on Windows. I found a solution at [style9-webpack](https://github.com/SukkaW/style9-webpack) (https://github.com/SukkaW/style9-webpack/issues/1) and made another Next.js plugin for vanilla-extract that works fine on Windows/Linux/macOS, if you're having a similar problem, you can try [@syfxlin/next-plugin-vanilla-extract](https://github.com/syfxlin/next-plugin-vanilla-extract).

**Comment by EugenePeter** [comment]
> I've recently been in the process of migrating my blog from Gatsby to Next.js, and I'm also experiencing problems with the Next.js RSC not outputting CSS on Windows. I found a solution at [style9-webpack](https://github.com/SukkaW/style9-webpack) ([SukkaW/style9-webpack#1](https://github.com/SukkaW/style9-webpack/issues/1)) and made another Next.js plugin for vanilla-extract that works fine on Windows/Linux/macOS, if you're having a similar problem, you can try [@syfxlin/next-plugin-vanilla-extract](https://github.com/syfxlin/next-plugin-vanilla-extract).

Using windows and can confirm that this worked

**Comment by askoufis** [comment]
Since the release of @syfxlin's [patch](https://github.com/vanilla-extract-css/vanilla-extract/pull/1180), hoping that the issues mentioned in this thread have been fixed. The windows-specific issue was being tracked in #1086, but I see that there have been some users reporting a similar issue on Ubuntu. If the issue is still occurring with the latest patch, please raise a separate issue instead of commenting on this one.

**Comment by thevisioner** [comment]
Long awaited fixes. 🙌 I can also confirm that it now works on a Windows system. Many thanks to the contributors! You're giving me a choice again. 😅
